src/Interpreters/DDLTask.cpp

```diff
@@ -74,6 +74,8 @@ void DDLLogEntry::setSettingsIfRequired(ContextPtr context)
 
     if (version >= SETTINGS_IN_ZK_VERSION)
         settings.emplace(context->getSettingsRef().changes());
+
+    max_query_size = context->getSettingsRef().max_query_size;
```

Comment:
This is only called on the initiator when enqueueing a query. The purpose of this method is to save changed settings from local context to the entry and propagate them to other hosts. It will not be initialized when parsing the entry